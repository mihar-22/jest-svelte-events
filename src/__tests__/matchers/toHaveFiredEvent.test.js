import '../../'
import { render, fireEvent } from '@testing-library/svelte'
import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredEvent', () => {
  test('passes when given an event that fired', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e1')
    const button = getByText('e1')
    await fireEvent.click(button)
    expect(component).toHaveFiredEvent('e1')
  })

  test('fails when given an event that did not fire', () => {
    const { component } = render(Comp)
    listen(component, 'e1')
    expect(() => {
      expect(component).toHaveFiredEvent('e1')
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredEvent', () => {
  test('passes when given an event that did not fire', () => {
    const { component } = render(Comp)
    listen(component, 'e1')
    expect(component).not.toHaveFiredEvent('e1')
  })

  test('fails when given an event that did fire', async () => {
    const { component, getByText } = render(Comp)
    listen(component, 'e1')
    const button = getByText('e1')
    await fireEvent.click(button)
    expect(() => {
      expect(component).not.toHaveFiredEvent('e1')
    }).toThrowErrorMatchingSnapshot()
  })
})
