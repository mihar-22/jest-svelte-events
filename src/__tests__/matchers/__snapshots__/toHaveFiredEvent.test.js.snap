// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredEvent fails when given an event that did fire 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredEvent[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired: [32m\\"e1\\"[39m
Recevied events: [31m[\\"e1\\"][39m"
`;

exports[`.toHaveFiredEvent fails when given an event that did not fire 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEvent[2m([22m[32mexpected[39m[2m)[22m

Expected to have fired: [32m\\"e1\\"[39m
Recevied events: [31m[][39m"
`;
